scipy/spatial/_geometric_slerp.py

```diff
@@ -28,10 +32,12 @@ def _geometric_slerp(start, end, t):
     return start * c[:, np.newaxis] + end * s[:, np.newaxis]
 
 
-def geometric_slerp(start,
-                    end,
-                    t,
-                    tol=1e-7):
+def geometric_slerp(
+    start: npt.ArrayLike,
+    end: npt.ArrayLike,
+    t: npt.ArrayLike,
+    tol: float = 1e-7,
+) -> np.ndarray:
```

Comment:
This can technically also be a list of arrays for as long as https://github.com/scipy/scipy/pull/14173 isn't merged.

Reply:
Considering the mypy error: on second thought, let's wait until aforementioned PR is merged.